python/cudf/cudf/core/column/datetime.py

```diff
@@ -219,6 +211,10 @@ def default_na_value(self):
                 "datetime column of {} has no NaN value".format(self.dtype)
             )
 
+    def binary_operator(self, op, rhs, reflect=False):
+        lhs, rhs = self, rhs
+        return binop(lhs, rhs, op=op, out_dtype=np.bool)
```

Comment:
Specifying the `out_dtype=np.bool` feels wrong here.

Reply:
Raising for all ops that don't do a comparison now.